torch/csrc/api/include/torch/nn/cloneable.h

```diff
@@ -48,7 +48,7 @@ class Cloneable : public virtual Module {
         "Are you sure you called register_parameter() inside reset() "
         "and not the constructor?");
     for (const auto& parameter : parameters_) {
-      auto data = autograd::Variable(*parameter).data().clone();
+      auto data = autograd::Variable(*parameter).clone();
```

Comment:
This looks wrong. `clone()` will preserve autograd metadata but `data()` did not. Now you're generating a possibly requires grad `data`.

Reply:
I think this is okay because `data` is being used in the `set_data(...)` call below, and `set_data(...)` ignores `data`'s AutogradMeta.